cmd/k8s-operator/sts.go

```diff
@@ -346,8 +357,8 @@ func (a *tailscaleSTSReconciler) reconcileSTS(ctx context.Context, logger *zap.S
 	ss.Spec.Template.Annotations = map[string]string{
 		"tailscale.com/operator-last-set-hostname": sts.Hostname,
 	}
-	if sts.TargetIP != "" {
-		ss.Spec.Template.Annotations["tailscale.com/operator-last-set-ip"] = sts.TargetIP
+	if sts.ClusterTargetIP != "" {
+		ss.Spec.Template.Annotations["tailscale.com/operator-last-set-ip"] = sts.ClusterTargetIP
```

Comment:
You want another similar annotation for egress

Reply:
I added this field to the `tailscaleSTSConfig` that will be populated with the value of the `tailscale.com/target-ip` annotation value

Reply:
>"tailscale.com/operator-last-set-ip"

is an annotation added by the controller so that the stateful set notices it and rolls out a new version of the proxy in case the value changes. You will need another annotation that matches the desired value of `TailscaleTargetIP`

Reply:
Ahh I misunderstood you- I thought you meant I was overriding an existing annotation. Added a `tailscale.com/operator-last-set-tailnet-ip` that will force the pod to restart if the target changes.